pkg/agent/util/net_windows.go

```diff
@@ -14,7 +14,281 @@
 
 package util
 
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/Microsoft/hcsshim"
+	ps "github.com/benmoss/go-powershell"
+	"github.com/benmoss/go-powershell/backend"
+	"github.com/containernetworking/plugins/pkg/ip"
+	"k8s.io/klog"
+)
+
 const (
-	LocalHNSNetwork     = "antrea-hnsnetwork"
 	ContainerVNICPrefix = "vEthernet"
+	HNSNetworkType      = "Transparent"
+	LocalHNSNetwork     = "antrea-hnsnetwork"
+	OVSExtensionID      = "583CC151-73EC-4A6A-8B47-578297AD7623"
 )
+
+// EnableHostInterface sets the specified interface status as UP.
+func EnableHostInterface(ifaceName string) error {
+	cmd := fmt.Sprintf("Enable-NetAdapter -InterfaceAlias %s", ifaceName)
+	return invokePSCommand(cmd)
+}
+
+// ConfigureInterfaceAddress adds IPAddress on the specified interface.
+func ConfigureInterfaceAddress(ifaceName string, ipConfig *net.IPNet) error {
+	ipStr := strings.Split(ipConfig.String(), "/")
+	cmd := fmt.Sprintf("New-NetIPAddress -InterfaceAlias %s -IPAddress %s -PrefixLength %s", ifaceName, ipStr[0], ipStr[1])
+	return invokePSCommand(cmd)
+}
+
+// EnableIPForwarding enables the IP interface to forward packets that arrive on this interface to other interfaces.
+func EnableIPForwarding(ifaceName string) error {
+	cmd := fmt.Sprintf("Set-NetIPInterface -InterfaceAlias %s -Forwarding Enabled", ifaceName)
+	return invokePSCommand(cmd)
+}
+
+func invokePSCommand(cmd string) error {
+	// Create a backend shell.
+	back := &backend.Local{}
+
+	// start a local powershell process
+	shell, err := ps.New(back)
+	if err != nil {
+		panic(err)
```

Comment:
why panic here instead of popping up the error?

Reply:
Done.